cell_classification/predict.py

```diff
@@ -0,0 +1,375 @@
+        for key in ["tp", "tn", "fp", "fn"]:  # sum fn, fp, tn, tp
+            avg_metrics[key].append(np.sum(metrics[key]))
+        for key in ["dataset", "imaging_platform", "marker", "threshold"]:  # copy strings
+            avg_metrics[key].append(metrics[key])
+    return avg_metrics
+
+
+def average_roc(roc_list):
+    """Average ROC curves
+    Args:
+        roc_list (list):
+            list of ROC curves
+    Returns:
+        tprs (np.array):
+            standardized true positive rates for each sample
+        mean_tprs (np.array):
+            mean true positive rates over all samples
+        std np.array:
+            standard deviation of true positive rates over all samples
+        base (np.array):
+            fpr values for interpolation
+        mean_thresh (np.array):
+            mean of the threshold values over all samples
+    """
+    base = np.linspace(0, 1, 101)
+    tpr_list = []
+    thresh_list = []
+    for i in range(len(roc_list["tpr"])):
+        tpr_list.append(np.interp(base, roc_list["fpr"][i], roc_list["tpr"][i]))
+        thresh_list.append(np.interp(base, roc_list["tpr"][i], roc_list["thresholds"][i]))
+
+    tprs = np.array(tpr_list)
+    thresh_list = np.array(thresh_list)
+    mean_thresh = np.mean(thresh_list, axis=0)
+    mean_tprs = tprs.mean(axis=0)
+    std = tprs.std(axis=0)
+    return tprs, mean_tprs, base, std, mean_thresh
+
+
+class HDF5Loader(object):
+    """HDF5 iterator for loading data from HDF5 files"""
+
+    def __init__(self, folder):
+        """Initialize HDF5 generator
+        Args:
+            folder (str):
+                path to folder containing HDF5 files
+        """
+        self.folder = folder
+        self.files = os.listdir(folder)
+        # filter out hdf files
+        self.files = [os.path.join(folder, f) for f in self.files if f.endswith(".hdf")]
+        self.file_idx = 0
+
+    def __len__(self):
+        return len(self.files)
+
+    def load_hdf(self, file):
+        """Load HDF5 file
+        Args:
+            file (str):
+                path to HDF5 file
+        Returns:
+            data (dict):
+                dictionary containing data from HDF5 file
+        """
+        out_dict = {}
+        with h5py.File(file, "r") as f:
+            keys = [key for key in f.keys() if key != "activity_df"]
+            for key in keys:
+                if isinstance(f[key][()], bytes):
+                    out_dict[key] = f[key][()].decode("utf-8")
+                else:
+                    out_dict[key] = f[key][()]
+        out_dict["activity_df"] = pd.read_hdf(file, key="activity_df", mode="r")
+        return out_dict
+
+    def __iter__(self):
+        self.file_idx = 0
+        return self
+
+    def __next__(self):
+        if self.file_idx >= len(self.files):
+            raise StopIteration
+        else:
+            self.file_idx += 1
+            return self.load_hdf(self.files[self.file_idx - 1])
+
+
+if __name__ == "__main__":
```

Comment:
Lets try and keep the code for running evaluations separate from the library of defined functions. This can go in `generate_accuracy_predictions.py` or something like that

Reply:
I think I'll add a folder "scripts" for this and put all the CLIs inside there.